pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
> Doesn't docker cmd require bash?

https://docs.docker.com/engine/reference/builder/#cmd

Docker's CMD in the shell form invokes `/bin/sh -c`.

Even though it's common to have the symlink, 

```console
$ ls -l /bin/{,ba}sh
-rwxr-xr-x. 1 root root 1071960 Aug 18 15:37 /bin/bash
lrwxrwxrwx. 1 root root       4 Aug 18 15:37 /bin/sh -> bash
```

... Bash starts in sh compatiblity mode when the base filename in `argv[0]` is `sh`, i.e. `/bin/sh` may be linked to `/bin/bash`, but that doesn't mean it acts like `/bin/bash`.


That said, I'm still favorable to using `/bin/sh -c`, going after what Docker does, and making our solution usable with a larger number of images.